hdk/virtual_machine/code.py

```diff
@@ -0,0 +1,171 @@
+"""Provides functions to translate Instruction into their assembly commands."""
+from collections.abc import Iterable, Iterator
+
+from hdk.virtual_machine.syntax import (
+    ArithmeticLogicalCommand,
+    Instruction,
+    MemoryTransferCommand,
+)
+
+_BINARY_TABLE = {
+    "add": "M=D+M",
+    "sub": "M=M-D",
+    "and": "M=D&M",
+    "or": "M=D|M",
+}
+
+_COMPARISON_TABLE = {
+    "eq": "JEQ",
+    "lt": "JLT",
+    "gt": "JGT",
+}
+
+_SEGMENT_TABLE = {
+    "local": "LCL",
+    "argument": "ARG",
+    "this": "THIS",
+    "that": "THAT",
+}
+
+
+def translate_stack_instruction(
```

Comment:
Rename to translate_arithmetic_locical_command. Change parameter name to command instead of stack_instruction.